compiler/pkg/openapi_generator/check-backward-compatibility.go

```diff
@@ -0,0 +1,142 @@
+package openapi_generator
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	log "github.com/sirupsen/logrus"
+
+	nexus_compare "github.com/vmware-tanzu/graph-framework-for-microservices/common-library/pkg/nexus-compare"
+	extensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+)
+
+func splitCRDs(content []byte) []string {
+	return strings.Split(string(content), "---")
+}
+
+func compareCRDs(inCompatibleCRDs []*bytes.Buffer, existingCRDName, existingCRDContent string, newCRDContent []byte) ([]*bytes.Buffer, error) {
+	newCRDParts := splitCRDs(newCRDContent)
+	for _, newCRDPart := range newCRDParts {
```

Comment:
This method needs a check for newCRD not found. 
The check in the caller does not catch it ?

Reply:
Modified code to throw error /fail the build when CRD content does not exist and CRD name not matched/found